### DISCUSSION TITLE
"GDALImageLayer FAILED" on osgEarth 3.2.0 sdk

### DISCUSSION THREAD

**Discussion by ghost** [pelicanmapping/osgearth#2017] [Support Q&A]
By following the procedure http://docs.osgearth.org/en/latest/build.html, the compilation and the installation go well. 

When I run osgearth_viewer with the example "simple.earth", I get the following error:

_[osgEarth]* [ShaderLoader] No source found for "RexEngine.SDK.glsl"
[osgEarth]* [ShaderLoader] No source found for "RexEngine.vert.glsl"
[osgEarth]* [ShaderLoader] No source found for "RexEngine.elevation.glsl"
[osgEarth]* [ShaderLoader] No source found for "RexEngine.NormalMap.glsl"
[osgEarth]* [ShaderLoader] No source found for "RexEngine.Morphing.glsl"
[osgEarth]* [ShaderLoader] No source found for "RexEngine.ImageLayer.glsl"
[osgEarth]* [ShaderLoader] No source found for "PBR.glsl"
[osgEarth]* [ShaderLoader] No source found for "LogDepthBuffer.VertOnly.glsl"
[osgEarth]* [ShaderLoader] No source found for "Text.glsl"
[osgEarth]* [ShaderLoader] No source found for "Text.glsl"
VERTEX glCompileShader "oe_logDepth_vert" FAILED
VERTEX Shader "oe_logDepth_vert" infolog:
WARNING: 0:5: '#extension' :  'GL_ARB_gpu_shader_int64' is not supported
ERROR: 0:10: 'gl_ProjectionMatrix' : undeclared identifier
ERROR: 0:10: 'gl_ProjectionMatrix' : left of '[' is not of type array, matrix, or vector
ERROR: 0:10: '' : left of '[' is not of type array, matrix, or vector
ERROR: 0:13: 'inverse' : no matching overloaded function found (using implicit conversion)
ERROR: 0:13: 'inverse' : function is not known
ERROR: 0:12: '=' :  cannot convert from 'const float' to '4X4 matrix of highp float'

VERTEX glCompileShader "oe_rex_init_model" FAILED
VERTEX Shader "oe_rex_init_model" infolog:
WARNING: 0:5: '#extension' :  'GL_ARB_gpu_shader_int64' is not supported
ERROR: 0:26: 'gl_MultiTexCoord0' : undeclared identifier
ERROR: 0:26: 'assign' :  cannot convert from 'highp float' to '4-component vector of highp float'

VERTEX glCompileShader "oe_rex_normalMapVS" FAILED
VERTEX Shader "oe_rex_normalMapVS" infolog:
WARNING: 2:5: '#extension' :  'GL_ARB_gpu_shader_int64' is not supported
ERROR: 2:34: 'gl_NormalMatrix' : undeclared identifier

VERTEX glCompileShader "oe_rex_morph" FAILED
VERTEX Shader "oe_rex_morph" infolog:
WARNING: 2:5: '#extension' :  'GL_ARB_gpu_shader_int64' is not supported
ERROR: 2:91: 'gl_ModelViewMatrix' : undeclared identifier
ERROR: 2:121: 'gl_MultiTexCoord1' : undeclared identifier
ERROR: 2:121: 'xyz' : vector field selection out of range
ERROR: 2:122: 'constructor' : not enough data provided for construction
ERROR: 2:122: 'gl_MultiTexCoord2' : undeclared identifier
ERROR: 2:122: 'xyz' : vector field selection out of range
ERROR: 2:122: '=' :  cannot convert from 'highp float' to '3-component vector of highp float'

VERTEX glCompileShader "main(vertex)" FAILED
VERTEX Shader "main(vertex)" infolog:
WARNING: 0:4: '#extension' :  'GL_ARB_gpu_shader_int64' is not supported
ERROR: 0:51: 'gl_Vertex' : undeclared identifier
ERROR: 0:51: 'assign' :  cannot convert from 'highp float' to '4-component vector of highp float'
ERROR: 0:52: 'gl_Normal' : undeclared identifier
ERROR: 0:52: 'assign' :  cannot convert from 'highp float' to '3-component vector of highp float'
ERROR: 0:53: 'gl_Color' : undeclared identifier
ERROR: 0:53: 'assign' :  cannot convert from 'highp float' to '4-component vector of highp float'
ERROR: 0:57: 'gl_ModelViewMatrix' : undeclared identifier
ERROR: 0:59: 'gl_NormalMatrix' : undeclared identifier
ERROR: 0:65: 'gl_ProjectionMatrix' : undeclared identifier

glLinkProgram 0000022E2778F090"GDALImageLayer" FAILED
Program "GDALImageLayer" infolog:
Attached vertex shader is not compiled._

If I run osgearth_version, I get the version "osgEarth 3.2.0 build 131".

If I simply install osgearth as indicated in the documentation, http://docs.osgearth.org/en/latest/install.html, in this case osgearth_viewer works with the "simple.earth" example. If I run osgearth_version, I have the version "osgEarth 3.2.0 build 113".

So I compile directly the sources from the zip archive on the osgearth site (https://github.com/gwaldron/osgearth/zipball/osgearth-3.2) and from the build procedure http://docs.osgearth.org/en/latest/build.html. In this case, the "simple.earth" example opens correctly with osgearth_viewer.

Did I miss something, or is there an anomaly with build 131 ?

Environment : Windows 11 64 bits, CMake 3.22.2, Visual Studio 2022 (64 bits) - Version 17.1.1

**Comment by ghost** [comment]
I looked at the issue"can not load simple_model.earth #1872" and I'm thinking there must be a link even if the message is not the same.

I rebuild sdk following documentation steps, and set GDAL_DATA, but I still have error message glLinkProgram 0000022E2778F090"GDALImageLayer" FAILED.

Version details with build 132
osgearth_version.exe --caps
[osgEarth]  [Capabilities] Capabilities:
[osgEarth]  [Capabilities]   osgEarth Version:  3.2.0 build 132
[osgEarth]  [Capabilities]   OSG Version:       3.6.5
[osgEarth]  [Capabilities]   GDAL Version:      3.4.1
[osgEarth]  [Capabilities]   GEOS Version:      3.10.0
[osgEarth]  [Capabilities]   GPU Vendor:        Intel
[osgEarth]  [Capabilities]   GPU Renderer:      Intel(R) UHD Graphics 630
[osgEarth]  [Capabilities]   GL/Driver Version: 4.6.0 - Build 30.0.101.1404
[osgEarth]  [Capabilities]   GL Core Profile:   no

**Reply by gwaldron** [reply]
Hello,
please set the environment variable OSG_NOTIFY_LEVEL=INFO, run again, and post the output. Thanks.

**Reply by ghost** [reply]
Hello,

the output is in attached file.

I saw these lines that seem to be related to the problem, but not sure:
 
Opened DynamicLibrary osgPlugins-3.6.5/osgdb_earth.dll
DynamicLibrary::failed loading "osgPlugins-3.6.5/osgdb_osgearth_gdalimage.dll"
DynamicLibrary::failed loading "osgEarthGdalimage.dll"
Opened DynamicLibrary osgPlugins-3.6.5/osgdb_osgearth_engine_rex.dll

Thanks.

[20220316_osgearth_log.txt](https://github.com/gwaldron/osgearth/files/8263160/20220316_osgearth_log.txt)
